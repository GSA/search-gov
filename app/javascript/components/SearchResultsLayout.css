.serp-result-wrapper{
  min-height: 35rem;
  padding-bottom: 3rem;
  padding-top: 3rem;
}

div[data-react-class="SearchResultsLayout"] {
  overflow-x: hidden;
}

a.logo-link, a.logo-link:hover, a.logo-link:visited {
  color: inherit;
  display: block;
  text-decoration: none;
}

.usa-logo__text {
  cursor: default;
}

.content-provider {
  margin-bottom: 1rem;
  padding-right: 2.5rem;
  text-align: right;
}

.bing {
  background: url('searches/bing_logo.png') right no-repeat;
  background-origin: content-box;
}

.bing > .powered-by {
  padding-right: 56px;
}

.bing > .engine {
  height:1px;
  left:-10000px;
  overflow:hidden;
  position:absolute;
  top:auto;
  width:1px;
}

@media (max-width: 767.98px){
  .serp-result-wrapper > .grid-container {
      padding-left: 0;
      padding-right: 0;
  }
}
