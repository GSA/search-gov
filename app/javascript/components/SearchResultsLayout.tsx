import React from 'react';
import { I18n } from 'i18n-js';

import './SearchResultsLayout.css';

import { Header } from './Header';
import { Facets } from './Facets/Facets';
import { SearchBar } from './SearchBar/SearchBar';
import { Results } from './Results/Results';
import { Footer } from './Footer/Footer';
import { Identifier } from './Identifier/Identifier';
import { LanguageContext } from '../contexts/LanguageContext';

export interface NavigationLink {
  active: boolean; label: string; url: string, facet: string;
}

interface SearchResultsLayoutProps {
  resultsData?: {
    totalPages: number;
    unboundedResults: boolean;
    results: {
      title: string;
      url: string;
      description: string;
      updatedDate?: string;
      publishedDate?: string;
      thumbnailUrl?: string;
      youtube?: boolean;
      youtubePublishedAt?: string;
      youtubeThumbnailUrl?: string;
      youtubeDuration?: string;
    }[] | null;
  } | null;
  additionalResults?: {
    recommendedBy: string;
    newNews?: {
      title: string,
      link: string,
      description: string,
      publishedAt: string
    }[];
    oldNews?: {
      title: string,
      link: string,
      description: string,
      publishedAt: string
    }[];
    textBestBets?: {
      title: string;
      url: string;
      description: string;
    }[];
    graphicsBestBet?: {
      title: string;
      titleUrl?: string;
      imageUrl?: string;
      imageAltText?: string;
      links?: {
        title: string;
        url: string;
      }[];
    };
    jobs?: {
      positionTitle: string;
      positionUri: string;
      positionLocationDisplay: string;
      organizationName: string;
      minimumPay: number;
      maximumPay: number;
      rateIntervalCode: string;
      applicationCloseDate: string;
    }[];
    youtubeNewsItems?: {
      link: string;
      title: string;
      description: string;
      publishedAt: string;
      youtubeThumbnailUrl: string;
      duration: string;
    }[];
    federalRegisterDocuments?: {
      commentsCloseOn: string | null;
      contributingAgencyNames: string[];
      documentNumber: string;
      documentType: string;
      endPage: number;
      htmlUrl: string;
      pageLength: number;
      publicationDate: string;
      startPage: number;
      title: string;
    }[];
    healthTopic?: {
      title: string;
      description: string;
      url: string;
      studiesAndTrials?: {
        title: string;
        url: string;
      }[];
      relatedTopics?: {
        title: string;
        url: string;
      }[];
    };
  } | null;
  vertical: string;
  params?: {
    query?: string
  };
  translations: {
    en?: { noResultsForAndTry: string }
  };
  currentLocale?: string;
  relatedSites?: {
    label: string;
    link: string;
  }[];
  alert?: {
    title: string;
    text: string;
  };
  spellingSuggestion?: {
    suggested: string;
    original: string;
  };
  navigationLinks: NavigationLink[];
  extendedHeader: boolean;
  fontsAndColors: {
    headerLinksFontFamily: string;
  };
  footerLinks?: {
    title: string,
    url: string
  }[];
  identifierContent?: {
    domainName: string | null;
    parentAgencyName: string | null;
    parentAgencyLink: string | null;
  };
  identifierLinks?: {
    title: string,
    url: string
  }[] | null;
  relatedSearches?: { label: string; link: string; }[];
  newsLabel?: {
    newsAboutQuery: string;
    results: {
      title: string;
      feedName: string,
      publishedAt: string
    }[] | null;
  } | null;
  relatedSitesDropdownLabel?: string;
  noResultsMessage?: {
    text: string;
    urls: {
      title: string;
      url: string;
    }[]
  };
}

// To be updated
const getAffiliateTitle = (): string => {
  return 'Search.gov';
};

const isBasicHeader = (extendedHeader: boolean): boolean => {
  return !extendedHeader;
};

const videosUrl = (links: NavigationLink[]) => links.find((link) => link.facet === 'YouTube')?.url ;

const SearchResultsLayout = ({ resultsData, additionalResults, vertical, params = {}, translations, currentLocale = 'en', relatedSites = [], extendedHeader, footerLinks, fontsAndColors, newsLabel, identifierContent, identifierLinks, navigationLinks, relatedSitesDropdownLabel = '', alert, spellingSuggestion, relatedSearches, noResultsMessage }: SearchResultsLayoutProps) => {
  const i18n = new I18n(translations);
  i18n.defaultLocale = 'en';
  i18n.enableFallback = true;
  i18n.locale = currentLocale;
  return (
    <LanguageContext.Provider value={i18n}>
      <Header 
        title={getAffiliateTitle()}
        isBasic={isBasicHeader(extendedHeader)}
        fontsAndColors={fontsAndColors}
      />
     
      <div className="usa-section serp-result-wrapper">
        <Facets />

        <SearchBar query={params.query} relatedSites={relatedSites} navigationLinks={navigationLinks} relatedSitesDropdownLabel={relatedSitesDropdownLabel} alert={alert}/>

        {/* This ternary is needed to handle the case when Bing pagination leads to a page with no results */}
        {resultsData ? (
          <Results 
            results={resultsData.results}
            vertical={vertical}
            totalPages={resultsData.totalPages}
            query={params.query}
            unboundedResults={resultsData.unboundedResults}
            additionalResults={additionalResults}
            newsAboutQuery={newsLabel?.newsAboutQuery}
            spellingSuggestion={spellingSuggestion}
            videosUrl= {videosUrl(navigationLinks)}
            relatedSearches = {relatedSearches}
            noResultsMessage = {noResultsMessage}
          />) : params.query ? (
          <Results 
            vertical={vertical}
            totalPages={null}
            query={params.query}
            unboundedResults={true}
            noResultsMessage = {noResultsMessage}
          />) : <></>}
      </div>

      <Footer 
        footerLinks={footerLinks}
      />
      <Identifier
        identifierContent={identifierContent}
        identifierLinks={identifierLinks}
      />
    </LanguageContext.Provider>
  );
};

export default SearchResultsLayout;
