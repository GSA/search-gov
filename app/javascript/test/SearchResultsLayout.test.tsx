import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import React from 'react';
import SearchResultsLayout from '../components/SearchResultsLayout';

describe('SearchResultsLayout', () => {
  it('shows a message when there are no results', () => {
    render(<SearchResultsLayout params="foo" results={[]} vertical="web" />);
    const message = screen.getByText(/Please enter a search term in the box above./i);
    expect(message).toBeInTheDocument();
  });

  it('renders all relevant links', () => {
    render(<SearchResultsLayout params="foo" results={[]} vertical="web" />);
    const everything = screen.getByText(/Everything/i);
    const news = screen.getByText(/News/i);
    const images = screen.getByText(/Images/i);
    const videos = screen.getByText(/Videos/i);
    expect(everything).toBeInTheDocument();
    expect(news).toBeInTheDocument();
    expect(images).toBeInTheDocument();
    expect(videos).toBeInTheDocument();
  });

  it('renders search results', () => {
    const results = [{ 'title': 'test result 1', 'unescapedUrl': 'https://www.search.gov', 'content': 'result body' }];
    render(<SearchResultsLayout params="foo" results={results} vertical="web" />);
    const resultTitle = screen.getByText(/test result 1/i);
    const resultUrl = screen.getByText(/https\:\/\/www.search.gov/i);
    const resultBody = screen.getByText(/result body/i);
    expect(resultTitle).toBeInTheDocument();
    expect(resultUrl).toBeInTheDocument();
    expect(resultBody).toBeInTheDocument();
  });

  it('renders image search results', () => {
    const results = [{ 'title': 'test result 1', 'thumbnail': { 'url': 'https://www.search.gov/test_image.png' } }];
    render(<SearchResultsLayout params="foo" results={results} vertical="image" />);
    const resultTitle = screen.getByText(/test result 1/i);
    const img = [...document.getElementsByClassName("result-image")].pop() as HTMLImageElement;
    expect(resultTitle).toBeInTheDocument();
    expect(img).toHaveAttribute('src', 'https://www.search.gov/test_image.png');
  });
});



