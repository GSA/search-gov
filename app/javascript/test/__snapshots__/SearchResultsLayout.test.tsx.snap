// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultsLayout renders basic header styles properly 1`] = `
<head>
  <style
    data-styled="active"
    data-styled-version="6.1.1"
  >
    .fSIIQD{color:#C000FE!important;}
    .iEGEJu{background-color:#4a402b;}
    .iEGEJu .usa-nav__primary,.iEGEJu .usa-nav__secondary-links{font-family:"Georgia","Cambria","Times New Roman","Times",serif;}
    .iEGEJu a.usa-nav__link{color:#594973;}
    .iEGEJu .usa-nav__secondary-item&gt;a{color:#c8155d;}
    .iEGEJu button.usa-menu-btn{background-color:#cfcd03;}
    .iEGEJu button.usa-menu-btn:hover{background-color:#9d9b02;}
    .fLKlZC .usa-banner__header,.fLKlZC .usa-banner__button-text{background-color:#643617;color:#dacb1b;}
    .fLKlZC .usa-banner__button::after,.fLKlZC .usa-banner__button[aria-expanded=true]:hover::after{background-color:#dacb1b;}
    .eaGpdP li.usa-nav__primary-item:not(li.usa-nav__submenu-item)&gt;a,.eaGpdP .usa-nav__primary&gt;.usa-nav__primary-item button[aria-expanded=false]{color:#aea9f7!important;}
    .eaGpdP .usa-current::after,.eaGpdP .usa-nav__primary&gt;.usa-nav__primary-item button[aria-expanded=true]{background-color:#aea9f7!important;}
    .eaGpdP .vertical-wrapper .usa-nav__submenu{background-color:#aea9f7!important;}
    .ckSPLK{font-family:"Helvetica Neue","Helvetica","Roboto","Arial",sans-serif;}
    .ckSPLK .result-title-link&gt;.result-title-label{color:#33f0aa!important;}
    .ckSPLK .result-title-link:visited&gt;.result-title-label{color:#4a97ad!important;}
    .ckSPLK .result-desc&gt;p{color:#2bd4c7!important;}
    .ckSPLK .result-url-text{color:#475830!important;}
    .eKmfZT{font-family:"Helvetica Neue","Helvetica","Roboto","Arial",sans-serif;background-color:#5fcfc5;}
    .eKmfZT .usa-footer__return-to-top&gt;a,.eKmfZT a.usa-footer__primary-link{color:#46f966;}
    .hsppyX{background-color:#be1c21;font-family:"Public Sans Web"!important;}
    .hsppyX .usa-identifier__container,.hsppyX .usa-identifier__container&gt;a,.hsppyX .usa-identifier__identity-disclaimer,.hsppyX .usa-identifier__identity-disclaimer&gt;a{color:#f48a4c;}
    .hsppyX .usa-identifier__required-links-item&gt;a,.hsppyX .usa-identifier__identity-domain{color:#5d5a6f;}
    .serp-result-wrapper{background-color:#761816;}
    .usa-button{background-color:#cfcd03;}
    .usa-button:hover{background-color:#9d9b02;}
  </style>
</head>
`;

exports[`SearchResultsLayout renders extended header styles properly 1`] = `
@media (max-width: 63.99em) {

}

<head>
  <style
    data-styled="active"
    data-styled-version="6.1.1"
  >
    .fSIIQD{color:#C000FE!important;}
    .frwUCo{background-color:#4a402b;}
    .frwUCo .usa-nav__primary a{font-family:"Public Sans Web"!important;font-weight:bold!important;}
    .frwUCo .usa-nav__secondary{font-family:"Georgia","Cambria","Times New Roman","Times",serif;}
    .frwUCo .usa-nav{background-color:#83df0a;}
    .frwUCo a.usa-nav__link{color:#594973!important;}
    .frwUCo .usa-nav__secondary-item&gt;a{color:#c8155d;}
    @media (max-width: 63.99em){.frwUCo .usa-nav__secondary-item&gt;a{color:#594973;}}
    .frwUCo button.usa-menu-btn{background-color:#cfcd03;}
    .frwUCo button.usa-menu-btn:hover{background-color:#9d9b02;}
    .fLKlZC .usa-banner__header,.fLKlZC .usa-banner__button-text{background-color:#643617;color:#dacb1b;}
    .fLKlZC .usa-banner__button::after,.fLKlZC .usa-banner__button[aria-expanded=true]:hover::after{background-color:#dacb1b;}
    .eaGpdP li.usa-nav__primary-item:not(li.usa-nav__submenu-item)&gt;a,.eaGpdP .usa-nav__primary&gt;.usa-nav__primary-item button[aria-expanded=false]{color:#aea9f7!important;}
    .eaGpdP .usa-current::after,.eaGpdP .usa-nav__primary&gt;.usa-nav__primary-item button[aria-expanded=true]{background-color:#aea9f7!important;}
    .eaGpdP .vertical-wrapper .usa-nav__submenu{background-color:#aea9f7!important;}
    .ckSPLK{font-family:"Helvetica Neue","Helvetica","Roboto","Arial",sans-serif;}
    .ckSPLK .result-title-link&gt;.result-title-label{color:#33f0aa!important;}
    .ckSPLK .result-title-link:visited&gt;.result-title-label{color:#4a97ad!important;}
    .ckSPLK .result-desc&gt;p{color:#2bd4c7!important;}
    .ckSPLK .result-url-text{color:#475830!important;}
    .eKmfZT{font-family:"Helvetica Neue","Helvetica","Roboto","Arial",sans-serif;background-color:#5fcfc5;}
    .eKmfZT .usa-footer__return-to-top&gt;a,.eKmfZT a.usa-footer__primary-link{color:#46f966;}
    .hsppyX{background-color:#be1c21;font-family:"Public Sans Web"!important;}
    .hsppyX .usa-identifier__container,.hsppyX .usa-identifier__container&gt;a,.hsppyX .usa-identifier__identity-disclaimer,.hsppyX .usa-identifier__identity-disclaimer&gt;a{color:#f48a4c;}
    .hsppyX .usa-identifier__required-links-item&gt;a,.hsppyX .usa-identifier__identity-domain{color:#5d5a6f;}
    .serp-result-wrapper{background-color:#761816;}
    .usa-button{background-color:#cfcd03;}
    .usa-button:hover{background-color:#9d9b02;}
  </style>
</head>
`;

exports[`SearchResultsLayout renders search with results styles properly 1`] = `
@media (max-width: 63.99em) {

}

<head>
  <style
    data-styled="active"
    data-styled-version="6.1.1"
  >
    .fSIIQD{color:#C000FE!important;}
    .frwUCo{background-color:#4a402b;}
    .frwUCo .usa-nav__primary a{font-family:"Public Sans Web"!important;font-weight:bold!important;}
    .frwUCo .usa-nav__secondary{font-family:"Georgia","Cambria","Times New Roman","Times",serif;}
    .frwUCo .usa-nav{background-color:#83df0a;}
    .frwUCo a.usa-nav__link{color:#594973!important;}
    .frwUCo .usa-nav__secondary-item&gt;a{color:#c8155d;}
    @media (max-width: 63.99em){.frwUCo .usa-nav__secondary-item&gt;a{color:#594973;}}
    .frwUCo button.usa-menu-btn{background-color:#cfcd03;}
    .frwUCo button.usa-menu-btn:hover{background-color:#9d9b02;}
    .fLKlZC .usa-banner__header,.fLKlZC .usa-banner__button-text{background-color:#643617;color:#dacb1b;}
    .fLKlZC .usa-banner__button::after,.fLKlZC .usa-banner__button[aria-expanded=true]:hover::after{background-color:#dacb1b;}
    .eaGpdP li.usa-nav__primary-item:not(li.usa-nav__submenu-item)&gt;a,.eaGpdP .usa-nav__primary&gt;.usa-nav__primary-item button[aria-expanded=false]{color:#aea9f7!important;}
    .eaGpdP .usa-current::after,.eaGpdP .usa-nav__primary&gt;.usa-nav__primary-item button[aria-expanded=true]{background-color:#aea9f7!important;}
    .eaGpdP .vertical-wrapper .usa-nav__submenu{background-color:#aea9f7!important;}
    .ckSPLK{font-family:"Helvetica Neue","Helvetica","Roboto","Arial",sans-serif;}
    .ckSPLK .result-title-link&gt;.result-title-label{color:#33f0aa!important;}
    .ckSPLK .result-title-link:visited&gt;.result-title-label{color:#4a97ad!important;}
    .ckSPLK .result-desc&gt;p{color:#2bd4c7!important;}
    .ckSPLK .result-url-text{color:#475830!important;}
    .eKmfZT{font-family:"Helvetica Neue","Helvetica","Roboto","Arial",sans-serif;background-color:#5fcfc5;}
    .eKmfZT .usa-footer__return-to-top&gt;a,.eKmfZT a.usa-footer__primary-link{color:#46f966;}
    .hsppyX{background-color:#be1c21;font-family:"Public Sans Web"!important;}
    .hsppyX .usa-identifier__container,.hsppyX .usa-identifier__container&gt;a,.hsppyX .usa-identifier__identity-disclaimer,.hsppyX .usa-identifier__identity-disclaimer&gt;a{color:#f48a4c;}
    .hsppyX .usa-identifier__required-links-item&gt;a,.hsppyX .usa-identifier__identity-domain{color:#5d5a6f;}
    .serp-result-wrapper{background-color:#761816;}
    .usa-button{background-color:#cfcd03;}
    .usa-button:hover{background-color:#9d9b02;}
  </style>
</head>
`;
